packages/config/lib/resolve-sets.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const assert = require('assert');
+
+const _ = {
+    uniqWith: require('lodash.uniqwith'),
+    isEqual: require('lodash.isequal')
+};
+
+// TODO: cache
+module.exports = function resolveSets(sets) {
+    return Object.keys(sets).reduce((acc, setName) => {
+        acc[setName] = _.uniqWith(resolveSet(sets[setName], setName, sets), _.isEqual);
+        return acc;
+    }, {});
+}
+
+function resolveSet(setData, setName, sets) {
+    if (typeof setData !== 'string') {
+        return Array.isArray(setData) ? setData : [setData];
+    }
+
+    return setData.split(' ').reduce((setDataAcc, layerStr) => {
+        if (!layerStr.includes('@')) {
+            setDataAcc.push({ layer: layerStr });
+            return setDataAcc;
+        }
+
+        const layerArr = layerStr.split('@');
+        let layerName = layerArr[0];
+        let libName = layerArr[1];
+
+        if (!layerName) {
+            const layerNameArr = libName.split('/');
+            libName = layerNameArr.shift();
```

Comment:
what is it about? `layer1@lib/layer2` ? Can you explain the difference of layer1 and layer2 ?

Reply:
`set1@lib` — means use all layers from set1 in lib
`@lib/layer2` — means use layer2 from lib
`set1@lib/layer2` — idk